rdr_service/dao/study_nph_dao.py

```diff
@@ -1,11 +1,493 @@
-from rdr_service.dao.base_dao import BaseDao
-from rdr_service.model.study_nph import Participant
+from protorpc import messages
+from typing import Tuple, Dict, List, Any
+from werkzeug.exceptions import BadRequest, NotFound
+import json
+from types import SimpleNamespace as Namespace
+from sqlalchemy.orm import Query, aliased
+from sqlalchemy import exc
 
+from rdr_service.model.study_nph import (
+    StudyCategory, Participant, Site, Order, OrderedSample, SampleUpdate, BiobankFileExport, SampleExport
+)
+from rdr_service.dao.base_dao import BaseDao, UpdatableDao
+import logging
+
+class OrderStatus(messages.Enum):
+    """A status reflecting the NPH order of the participant"""
+
+    RESTORED = 1
+    CANCELED = 2
 
-class NphParticipantDao(BaseDao):
 
+class NphParticipantDao(BaseDao):
     def __init__(self):
         super(NphParticipantDao, self).__init__(Participant)
 
-    def get_id(self, obj: Participant):
+    def get_id(self, session, nph_participant_id: str) -> int:
+        participant_id = self.convert_id(nph_participant_id)
+        query = Query(Participant)
+        query.session = session
+        result = query.filter(Participant.id == participant_id).first()
+        if result:
+            return result.id
+        else:
+            raise NotFound(f"Participant ID not found : {participant_id}")
+
+    def get_participant(self, nph_participant_id: str, session) -> Participant:
+        participant_id = self.convert_id(nph_participant_id)
+        query = Query(Participant)
+        query.session = session
+        result = query.filter(Participant.id == participant_id).first()
+        if result:
+            return result
+        else:
+            raise NotFound(f"Participant not found : {participant_id}")
+
+    def check_participant_exist(self, nph_participant_id: str, session) -> bool:
+        participant_id = self.convert_id(nph_participant_id)
+        query = Query(Participant)
+        query.session = session
+        result = query.filter(Participant.id == participant_id).first()
+        if result:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def convert_id(nph_participant_id: str) -> int:
+        return int(nph_participant_id[3:])
```

Comment:
We're expecting NPH ids to be prefixed by "1000".
```suggestion
        return int(nph_participant_id[4:])
```

Reply:
Also, this method should ensure that the participant id isn't an empty string and has atleast the needed number of characters.

Reply:
updated to slice to index #4